src/main/java/co/cask/wrangler/internal/TextSpecification.java

```diff
@@ -66,8 +69,33 @@
   // directives for wrangling.
   private String[] directives;
 
+  // Mapping of specification formats.
+  Map<String, String> formats = new HashMap<>();
+
   public TextSpecification(String[] directives) {
     this.directives = directives;
+
+    // Add all the usages.
+    formats.put("set format", "set format [csv|json] <delimiter> <skip empty lines>");
```

Comment:
where is this being used? Also, is there a way to add a method to each directive to define this so that when someone adds a new directive, they don't have to make updates in multiple places?